group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/OptimizedUniformAssignmentBuilder.java

```diff
@@ -18,29 +18,18 @@
 
 import org.apache.kafka.common.Uuid;
 import org.apache.kafka.server.common.TopicIdPartition;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import java.util.ArrayList;
 import java.util.Collections;
-import java.util.Comparator;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
-import java.util.Queue;
 import java.util.Set;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
-
-import static java.lang.Math.min;
 
 /**
- * The optimized uniform assignment builder is used to generate the target assignment for a consumer group with
+ * The homogenous uniform assignment builder is used to generate the target assignment for a consumer group with
```

Comment:
Are we trying to rename the assignor to homogeneous and heterogeneous instead of optimized and general?

Reply:
I would like too. I initially renamed the homogeneous one in this PR but @jeffkbkim asked to do it as a follow-up in order to simplify the diff.

Reply:
I feel like homogeneous and heterogeneous are harder names from a user pov, I think we should keep it as is

Reply:
Interesting… So you’re fine with adding those names to the public interface (what the end user really see) but not to follow the same naming scheme for the internal classes (that the end user don’t see at all). In my opinion, it makes sense to align as it will make it more coherent.

Reply:
That was for the subscription type right? Homogeneous uniform assignment builder is kind of a mouthful. 
I used the wrong term, by user I meant anyone looking at the code.